### PR TITLE
Feat(PE-3642): edit undername interaction

### PR DESCRIPTION
**Pull request by atticusofsparta**
https://sonar.warp.cc/#/app/interaction/PFkc3It3yf48dMWmHi1NuzJSwCYTPPtybDuboLxGNYo

### CODE REVIEW

File: src/components/layout/Undernames/Undernames.tsx [hunk 1]
```diff
@@ -476,7 +561,11 @@ function Undernames() {
                       setValue={(e) => {
                         e ? setTTL(+e) : setTTL(undefined);
                       }}
-                      validationPredicates={{}}
+                      validityCallback={(isValid) => setChangesValid(isValid)}
+                      validationPredicates={{
+                        [VALIDATION_INPUT_TYPES.VALID_TTL]: (ttlSeconds) =>
+                          arweaveDataProvider.validateTTLSeconds(+ttlSeconds),
```
Comment (dtfiedler) [inline src/components/layout/Undernames/Undernames.tsx:567]: this is just a util method, it doesn't require an API call so no need to attach to this provider. move it to a separate util function

File: src/services/arweave/SimpleArweaveDataProvider.ts [hunk 2]
```diff
@@ -118,6 +122,23 @@ export class SimpleArweaveDataProvider implements ArweaveDataProvider {
...
+      }
+      if (ttl > MAX_TTL_SECONDS) {
+        throw new Error(
+          `${ttl} is more than the maximum ttlSeconds requirement of ${MAX_TTL_SECONDS}`,
+        );
+      }
+    } catch (error: any) {
+      throw error;
+    }
+  }
```
Comment (dtfiedler) [inline src/services/arweave/SimpleArweaveDataProvider.ts:140]: see comment above - this provider is meant for API calls to arweave/warp/cache and not util validations. move to separate function

File: src/utils/transactionUtils/transactionUtils.ts [hunk 3]
```diff
@@ -572,3 +576,15 @@ export function getAssociatedNames(
     })
     .filter((n) => !!n);
 }
+export async function validateTTLSeconds(ttl: number): Promise<void> {
```
Comment (dtfiedler) [inline src/utils/transactionUtils/transactionUtils.ts:579]: nit: this isn't an async function so `async` and `Promise` are not needed 😄

Comment (atticusofsparta) [reply src/utils/transactionUtils/transactionUtils.ts:579]: yeah this was simply to make it compatible with the Promise<any> return type

Comment (atticusofsparta) [reply src/utils/transactionUtils/transactionUtils.ts]: changed to a manual promise

File: src/components/layout/Undernames/Undernames.tsx [hunk 4]
```diff
@@ -217,6 +247,62 @@ function Undernames() {
...
+              if (ttl < MIN_TTL_SECONDS) {
+                throw new Error(
+                  `${ttl} is less than the minimum ttlSeconds requirement of ${MIN_TTL_SECONDS}`,
+                );
+              }
+              if (ttl > MAX_TTL_SECONDS) {
+                throw new Error(
+                  `${ttl} is more than the maximum ttlSeconds requirement of ${MAX_TTL_SECONDS}`,
+                );
+              }
```
Comment (dtfiedler) [inline src/components/layout/Undernames/Undernames.tsx:268]: this should be able to use the util - no?

**Review by dtfiedler — APPROVED** [review]

**Review by dtfiedler — APPROVED** [review]

**Pull request merged**